### PR TITLE
feat(MultiTypeaheadSelect): Add MultiTypeaheadSelect to react-templates

### LINKED ISSUE #10733: MultiTypeAheadSelect - Add a react-template for multi type ahead select
**Is this a new component or an extension of an existing one?**
Could be an extension of TypeAheadSelect

**Describe the feature**
The doc page shows a multi type ahead select but the code is all in the demo. There is no template that shares reusable code for this component.

### PR DESCRIPTION
**Pull request by jeff-phillips-18**
**What**:
Closes #10733 

**Description**:
Adds a `MultiTypeaheadSelect` component to the @patternfly/react-templates package and a demo of the component for the docs apge.

### CODE REVIEW

File: packages/react-templates/src/components/Select/MultiTypeaheadSelect.tsx [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+  const [selected, setSelected] = React.useState<(string | number)[]>(
+    (initialOptions?.filter((o) => o.selected) ?? []).map((o) => o.value)
+  );
+  const [inputValue, setInputValue] = React.useState<string>();
+  const [selectOptions, setSelectOptions] = React.useState<MultiTypeaheadSelectOption[]>(initialOptions);
+  const [focusedItemIndex, setFocusedItemIndex] = React.useState<number | null>(null);
+  const [activeItemId, setActiveItemId] = React.useState<string | null>(null);
+  const textInputRef = React.useRef<HTMLInputElement>();
+
+  const NO_RESULTS = 'no results';
```
Comment (tlabaj) [inline packages/react-templates/src/components/Select/MultiTypeaheadSelect.tsx:76]: Can we add a prop so consumers an change this string?

**Review by tlabaj — CHANGES_REQUESTED** [review]

**Review by dlabaj — APPROVED** [review]
Other then titani's comment LGTM.

**Review by tlabaj — APPROVED** [review]

File: packages/react-templates/src/components/Select/MultiTypeaheadSelect.tsx (hunk 1 above)
Comment (tlabaj) [reply packages/react-templates/src/components/Select/MultiTypeaheadSelect.tsx:76]: Never mind.  It looks like that is not a display string.

File: packages/react-templates/src/components/Select/MultiTypeaheadSelect.tsx [hunk 2]
```diff
@@ -0,0 +1,348 @@
...
+    } else if (!inputValue) {
+      closeMenu();
+    }
+  };
+
+  const selectOption = (
+    _event: React.MouseEvent<Element, MouseEvent> | React.KeyboardEvent<HTMLInputElement> | undefined,
+    option: string | number
+  ) => {
+    const selections = selected.includes(option) ? selected.filter((o) => option === o) : [...selected, option];
```
Comment (kmcfaul) [inline packages/react-templates/src/components/Select/MultiTypeaheadSelect.tsx:147]: The true branch here should be `selected.filter((o) => option !== o)`. Currently it's resetting the selections array when an option is deselected.

**Review by kmcfaul — CHANGES_REQUESTED** [review]
Just one behavioral bug, everything else lgtm

File: packages/react-templates/src/components/Select/MultiTypeaheadSelect.tsx (hunk 2 above)
Comment (kmcfaul) [inline packages/react-templates/src/components/Select/MultiTypeaheadSelect.tsx:147]:
```suggestion
    const selections = selected.includes(option) ? selected.filter((o) => option !== o) : [...selected, option];
```

Comment (kmcfaul) [reply packages/react-templates/src/components/Select/MultiTypeaheadSelect.tsx:147]: The true branch here should be selected.filter((o) => option !== o). Currently it's resetting the selections array when an option is deselected.

**Review by jenny-s51 — APPROVED** [review]

**Review by kmcfaul — APPROVED** [review]

**Pull request merged**